src/ResourceManagement/CosmosDB/IndexingPolicyImpl.cs

```diff
@@ -0,0 +1,278 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Management.CosmosDB.Fluent.Models;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Management.CosmosDB.Fluent
+{
+
+    internal partial class IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> :
+        ChildResource<Models.IndexingPolicy, ParentImplT, IParentT>,
+        IndexingPolicy.Definition.IDefinition<DefinitionParentT>,
+        IndexingPolicy.Update.IUpdate<UpdateParentT>
+        where ParentImplT : IParentT, DefinitionParentT, UpdateParentT, IHasIndexingPolicy<ParentImplT>
+    {
+        public IndexingPolicyImpl(Models.IndexingPolicy inner, ParentImplT parent)
+            : base(inner, parent)
+        {
+        }
+
+        public override string Name()
+        {
+            return null;
+        }
+
+        public ParentImplT Attach()
+        {
+            return this.Parent.WithIndexingPolicy(this.Inner);
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithAutomatic(bool isAutomatic)
+        {
+            Inner.Automatic = isAutomatic;
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithIndexingMode(IndexingMode indexingMode)
+        {
+            Inner.IndexingMode = indexingMode;
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithoutIndexingMode()
+        {
+            Inner.IndexingMode = null;
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithIncludedPathsAppend(IList<IncludedPath> includedPaths)
+        {
+            if (Inner.IncludedPaths == null)
+            {
+                Inner.IncludedPaths = new List<IncludedPath>();
+            }
+
+            foreach (var includePath in includedPaths)
+            {
+                Inner.IncludedPaths.Add(includePath);
+            }
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithIncludedPathsReplace(IList<IncludedPath> includedPaths)
+        {
+            Inner.IncludedPaths = includedPaths;
```

Comment:
Better to make a copy of the List, in case user update the list later.

Reply:
done